### PR TITLE
Move static code outside of unciode.py.

### PR DESCRIPTION
**Pull request by clarfonthey**
This script in libstd_unicode is a mess and also contains code that shouldn't be output by a script, and instead just put in modules. So, this change does that.

### CODE REVIEW

File: src/libstd_unicode/tables.rs [hunk 1]
```diff
@@ -40,74 +25,8 @@ pub const UNICODE_VERSION: UnicodeVersion = UnicodeVersion {
...
-        match self.r1.get(c >> 6) {
-            Some(&child) => trie_range_leaf(c, self.r2[child as usize]),
-            None => false,
-        }
-    }
-}
-
 pub mod general_category {
-    pub const Cc_table: &'static super::SmallBoolTrie = &super::SmallBoolTrie {
+    pub const Cc_table: &SmallBoolTrie = &super::SmallBoolTrie {
```
Comment (kennytm) [inline src/libstd_unicode/tables.rs]:
```
[00:04:03] error[E0412]: cannot find type `SmallBoolTrie` in this scope
[00:04:03]   --> /checkout/src/libstd_unicode/tables.rs:29:26
[00:04:03]    |
[00:04:03] 29 |     pub const Cc_table: &SmallBoolTrie = &super::SmallBoolTrie {
[00:04:03]    |                          ^^^^^^^^^^^^^ not found in this scope
[00:04:03] help: possible candidate is found in another module, you can import it into scope
[00:04:03]    |
[00:04:03] 29 |     use bool_trie::SmallBoolTrie;
[00:04:03]    |
```

File: src/libstd_unicode/tables.rs [hunk 2]
```diff
@@ -120,7 +39,7 @@ pub mod general_category {
         Cc_table.lookup(c)
     }
 
-    pub const N_table: &'static super::BoolTrie = &super::BoolTrie {
+    pub const N_table: &BoolTrie = &BoolTrie {
```
Comment (kennytm) [inline src/libstd_unicode/tables.rs]:
```
[00:04:03] error[E0412]: cannot find type `BoolTrie` in this scope
[00:04:03]   --> /checkout/src/libstd_unicode/tables.rs:42:25
[00:04:03]    |
[00:04:03] 42 |     pub const N_table: &BoolTrie = &BoolTrie {
[00:04:03]    |                         ^^^^^^^^ not found in this scope
[00:04:03] help: possible candidate is found in another module, you can import it into scope
[00:04:03]    |
[00:04:03] 29 |     use bool_trie::BoolTrie;
[00:04:03]    |
[00:04:03] 
[00:04:03] error[E0422]: cannot find struct, variant or union type `BoolTrie` in this scope
[00:04:03]   --> /checkout/src/libstd_unicode/tables.rs:42:37
[00:04:03]    |
[00:04:03] 42 |     pub const N_table: &BoolTrie = &BoolTrie {
[00:04:03]    |                                     ^^^^^^^^ not found in this scope
[00:04:03] help: possible candidate is found in another module, you can import it into scope
[00:04:03]    |
[00:04:03] 29 |     use bool_trie::BoolTrie;
[00:04:03]    |
```

File: src/libstd_unicode/tables.rs [hunk 3]
```diff
@@ -212,13 +131,13 @@ pub mod general_category {
...
     pub fn N(c: char) -> bool {
-        super::trie_lookup_range_table(c, N_table)
+        N_table.lookup(c)
     }
 
 }
 
 pub mod derived_property {
-    pub const Alphabetic_table: &'static super::BoolTrie = &super::BoolTrie {
+    pub const Alphabetic_table: &BoolTrie = &BoolTrie {
```
Comment (kennytm) [inline src/libstd_unicode/tables.rs]: cannot find type `BoolTrie` in this scope

File: src/libstd_unicode/tables.rs [hunk 4]
```diff
@@ -397,10 +316,10 @@ pub mod derived_property {
     };
 
     pub fn Alphabetic(c: char) -> bool {
-        super::trie_lookup_range_table(c, Alphabetic_table)
+        Alphabetic_table.lookup(c)
     }
 
-    pub const Case_Ignorable_table: &'static super::BoolTrie = &super::BoolTrie {
+    pub const Case_Ignorable_table: &BoolTrie = &BoolTrie {
```
Comment (kennytm) [inline src/libstd_unicode/tables.rs]: etc.

File: src/libstd_unicode/tables.rs [hunk 5]
```diff
@@ -12,23 +12,8 @@
...
-    pub minor: u32,
-
-    /// Micro (or Update) version.
-    pub micro: u32,
-
-    // Private field to keep struct expandable.
-    _priv: (),
-}
+use version::UnicodeVersion;
+use bool_trie::{BoolTrie, SmallBoolTrie};
```
Comment (kennytm) [inline src/libstd_unicode/tables.rs]:
```
[00:04:03] error: unused import: `BoolTrie`
[00:04:03]   --> /checkout/src/libstd_unicode/tables.rs:16:17
[00:04:03]    |
[00:04:03] 16 | use bool_trie::{BoolTrie, SmallBoolTrie};
[00:04:03]    |                 ^^^^^^^^
[00:04:03]    |
```

**Comment by clarcharr** [comment]
@kennytm I actually just stumbled upon those errors this morning; I just pushed the fix.

File: src/libstd_unicode/version.rs [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+/// Represents a Unicode Version.
+///
+/// See also: <http://www.unicode.org/versions/>
+#[non_exhaustive]
```
Comment (alexcrichton) [inline src/libstd_unicode/version.rs]: Could this be omitted? It seems like there's not really much need to pull in a new unstable feature to this library

File: src/libstd_unicode/bool_trie.rs [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+/// exploits the fact that in reasonable sets, many such leaves can be shared.
+///
+/// In the third (0x10000..0x110000), each child of the root node represents a
+/// 4096-wide subrange, and the trie stores an 8-bit index into a 64-byte slice
+/// of a child tree. Each of these 64 bytes represents an index into the table
+/// of shared 64-bit leaf values. This exploits the sparse structure in the
+/// non-BMP range of most Unicode sets.
+pub struct BoolTrie {
+    // 0..0x800 (corresponding to 1 and 2 byte utf-8 sequences)
+    pub(crate) r1: [u64; 32],   // leaves
```
Comment (alexcrichton) [inline src/libstd_unicode/bool_trie.rs]: Can these just be `pub` as this module isn't exported anyway?

Comment (clarcharr) [reply src/libstd_unicode/bool_trie.rs]: Will do

File: src/libstd_unicode/version.rs (hunk 6 above)
Comment (clarcharr) [reply src/libstd_unicode/version.rs]: Will leave this to #47122 to fix and remove it from this PR.

**Comment by clarcharr** [comment]
Requested changes have been made.

**Pull request merged**